presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/PushAggregationIntoTableScan.java

```diff
@@ -139,14 +135,18 @@ public Result apply(AggregationNode node, Captures captures, Context context)
                 .collect(toImmutableList());
 
         List<ColumnHandle> groupByColumns = groupingKeys.stream()
-                .map(groupByColumn -> assignments.get(groupByColumn.getName()))
+                .map(tableScan.getAssignments()::get)
                 .collect(toImmutableList());
 
         Optional<AggregationApplicationResult<TableHandle>> aggregationPushdownResult = metadata.applyAggregation(
                 context.getSession(),
                 tableScan.getTable(),
                 aggregateFunctions,
-                assignments,
+                aggregateFunctions.isEmpty()
+                        ? Map.of()
+                        : tableScan.getAssignments()
+                        .entrySet().stream()
+                        .collect(toImmutableMap(entry -> entry.getKey().getName(), Entry::getValue)),
```

Comment:
I had this so initially, but the readability was similarly affected with variable or without.
However, having a variable is flaky, if the content is conditional -- note that previously we used those `assignments` instead of `tableScan.getAssignments` and now it would not be correct.